Manager can view the request created by direct report

Kaytbode opened the pull request:
**What does this PR do?**
Manager can view pending requests initiated by direct reports

**Description of Task to be completed?**
Retrieve from the database all pending requests for approval
Return results based on the requester's manager

**How should this be manually tested?**
npm run test

**Any background context you want to provide?**
postman's response
![reqresponse](https://user-images.githubusercontent.com/18212299/64501997-b93cd600-d2bb-11e9-883d-0252b925732d.JPG)

What are the relevant pivotal tracker stories?
#167730645

iakhator commented on src/database/seeders/20190908211400-users.js:
```diff
@@ -0,0 +1,44 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
```
attend to this.

iakhator commented on src/database/seeders/20190908211400-users.js:
```diff
@@ -0,0 +1,44 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
```
this

iakhator commented on src/database/seeders/20190908212920-travel_requests.js:
```diff
@@ -0,0 +1,37 @@
+module.exports = {
```
Try to exclude module in your eslint config

Kaytbode replied:
@iakhator I actually did, but i will have to include it in the file for houndci to stop screaming

iakhator commented on src/services/travelServices.js:
```diff
@@ -1,18 +1,20 @@
 import 'core-js/stable';
 import 'regenerator-runtime/runtime';
 import models from '../database/models';
+import { async } from 'regenerator-runtime/runtime';
```
all your hound errors should be fixed

chuksjoe commented on src/database/migrations/20190908184324-create-travel-request.js:
```diff
@@ -0,0 +1,93 @@
+          msg: 'Please provide reason for travel'
+        }
+      },
+      accommodation_id: {
```
I will suggest we use string datatype for our accommodation ID so that it can accommodate multi-city travel requests in which case we can separate the IDs with commas.

Kaytbode replied:
@chuksjoe I am not really getting the point. please could you be more explicit, maybe come up with a scenario?

Kaytbode commented:
> Please attend to the hound errors

Ok i will

chuksjoe commented on src/database/models/travel_request.js:
```diff
@@ -0,0 +1,33 @@
+      onDelete: 'CASCADE'
+    });
+
+    travel_request.belongsTo(models.accommodations, {
```
I don't think this association is necessary. Deleting the entries in the accommodation table shouldn't automatically delete any travel request.

Kaytbode replied:
@chuksjoe .. You are right about that.. but the association references a particular accommodation, and for the same reason you stated. if the accommodation detail is modified or deleted. **postgres will scream**. I think the application needs that

chuksjoe commented on src/services/travelServices.js:
```diff
@@ -1,18 +1,20 @@
+import { async } from 'regenerator-runtime/runtime';
 
-const { travels } = models;
+const { travels, travel_requests, departments, users } = models;
```
I don't think is ideal we having two travel request tables. I will suggest you modify the already existing table to fit into the task you are working on, and also change the name from travel to travel_request as agreed by the team.

Kaytbode replied:
@chuksjoe  that's fine, if we all agree it should be removed.

danoseun replied:
I thought we already agreed to have one request table?

chuksjoe commented on src/controllers/travelControllers.js:
```diff
@@ -21,4 +21,26 @@ export default {
+
+        const { manager } = req.params;
+
+        if (role == 'user') {
```
I don't think the double = for equality is ideal. Why not use the strict equality operator (===)?

Kaytbode commented:
@iakhator  i have fixed the hound issues..

Kaytbode commented:
I appreciate the reviews, and i have worked on them

@iakhator . the hound issues are as a result of the eslint version we are using **6.1.0** being higher than the one that is supported **5.7.0**.  
@chuksjoe travels table has been removed.

The tests are failing due to the email verification ultimatum reached.

chuksjoe commented:
Good morning @Kaytbode. Well done for the effort so far, I can see progress in the Hound report.
I will advise that you reduce the number of commits you have here. 14 is pretty much.
To reduce it to one commit, you can use `git rebase -i HEAD~<number-of-commits-to-squash>`.
Then, for further commits, you can use `git commit --amend` after you have staged the changes.
Please, endeavor to do this to reduce the headache encountered when any of us might want to rebase in the future.

chuksjoe commented:
Also, the current Hound errors `The configured eslint version is not supported.` most likely will be due to what you might have added to the `.eslintrc` file. I will advise you undo the modifications in the `.eslintrc` file and fix the errors it's flagging in your codes.

iakhator commented on src/controllers/authControllers.js:
```diff
@@ -8,32 +8,46 @@ import message from '../utils/messageUtils';
             delete data.password;
-            successResponseWithData(res, statusCode.created, message.signupSuccess(req.email), data);
+            successResponseWithData(res,
+              statusCode.created, message.signupSuccess(req.email), data);
```
There should always be a newline at the end of your file.

iakhator commented on src/controllers/authControllers.js:
```diff
@@ -8,32 +9,46 @@ import message from '../utils/messageUtils';
-            const { first_name, last_name, email, password } = req.body;
-            const userObj = { first_name, last_name, email, password, role: 'user'};
+            const {
+              first_name, last_name, email, password
```
2 line spacing

iakhator replied:
Inconsistent line spacing

iakhator commented on src/controllers/authControllers.js:
```diff
@@ -8,32 +9,46 @@ import message from '../utils/messageUtils';
     fbgooglesignup: async (req, res) => {
         try {
-            const { first_name, last_name, email, id, role } = req.user;
+            const {
```
4 line spacing.

iakhator commented on src/database/migrations/20190908184324-create-travel-request.js:
```diff
@@ -0,0 +1,93 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => {
```
you should remove the arrow function and write a proper function. You don't have to write all syntax as ES6. function is still a valid Javascript.

Kaytbode commented:
> sort out your spacing with the team.

Thank you for the review

Yes we agreed to 2 spaces, following the airbnb style guide

Kaytbode commented:
> Also, the current Hound errors `The configured eslint version is not supported.` most likely will be due to what you might have added to the `.eslintrc` file. I will advise you undo the modifications in the `.eslintrc` file and fix the errors it's flagging in your codes.

The modifications were what i added to the hound.yml file to support the eslint version we are using. However, i have reverted to what it was since it does not support latest eslint version.

Kaytbode commented:
@iakhator  Thank you for the reviews, and i have applied the feedback gotten.

@chuksjoe Thanks for your review. Feedback applied
